Analysis/Tasks/trackselection.cxx

```diff
@@ -74,25 +77,17 @@ struct TrackExtensionTask {
 
   void process(aod::Collision const& collision, aod::FullTracks const& tracks)
   {
-    float sinAlpha = 0.f;
-    float cosAlpha = 0.f;
-    float globalX = 0.f;
-    float globalY = 0.f;
-    float dcaXY = 0.f;
-    float dcaZ = 0.f;
-
     for (auto& track : tracks) {
-
-      sinAlpha = sin(track.alpha());
-      cosAlpha = cos(track.alpha());
-      globalX = track.x() * cosAlpha - track.y() * sinAlpha;
-      globalY = track.x() * sinAlpha + track.y() * cosAlpha;
-
-      dcaXY = track.charge() * sqrt(pow((globalX - collision.posX()), 2) +
-                                    pow((globalY - collision.posY()), 2));
-      dcaZ = track.charge() * sqrt(pow(track.z() - collision.posZ(), 2));
-
-      extendedTrackQuantities(dcaXY, dcaZ);
+      // FIXME: can we simplify this knowing that track is already at dca without copying code from TrackPar?
+      float magField = 5.0; // in kG (FIXME: get this from CCDB)
+      std::array<float, 2> dca{};
+      auto trackPar = getTrackParCov(track);
+      trackPar.propagateParamToDCA({collision.posX(), collision.posY(), collision.posZ()}, magField, &dca);
```

Comment:
Yes, either init ``dca`` with very large values or check the returned value of ``propagateParamToDCA`` and then set dca to large value.